src/main/java/taintedmagic/common/items/wand/foci/ItemFocusTaintedBlast.java

```diff
@@ -45,95 +47,119 @@ public void registerIcons(IIconRegister ir) {
         this.ornIcon = ir.registerIcon("taintedmagic:ItemFocusTaintedBlast_orn");
     }
 
-    public IIcon getFocusDepthLayerIcon(ItemStack s) {
+    @Override
+    public IIcon getFocusDepthLayerIcon(ItemStack stack) {
         return this.depthIcon;
     }
 
-    public IIcon getOrnament(ItemStack s) {
+    @Override
+    public IIcon getOrnament(ItemStack stack) {
         return this.ornIcon;
     }
 
-    public String getSortingHelper(ItemStack s) {
-        return "SHOCKWAVE" + super.getSortingHelper(s);
+    @Override
+    public String getSortingHelper(ItemStack stack) {
+        return "SHOCKWAVE" + super.getSortingHelper(stack);
     }
 
-    public int getFocusColor(ItemStack s) {
+    @Override
+    public int getFocusColor(ItemStack stack) {
         return 13107455;
     }
 
-    public AspectList getVisCost(ItemStack s) {
-        return costBase;
+    @Override
+    public AspectList getVisCost(ItemStack stack) {
+        AspectList list = costBase.copy();
+        if (TaintedMagicHelper.hasFocusUpgrade(stack, FocusUpgradeType.enlarge)) {
+            for (int i = 0; i < TaintedMagicHelper.getFocusLevelUpgrade(stack, FocusUpgradeType.enlarge); i++)
```

Comment:
Store result of helper call in variable (to prevent calling helper on every iteration).